android/src/main/java/io/github/tastelessjolt/flutterdynamicicon/MethodCallHandlerImpl.java

```diff
@@ -1,18 +1,30 @@
 package io.github.tastelessjolt.flutterdynamicicon;
 
+import android.content.Context;
+import android.content.pm.ActivityInfo;
+
 import io.flutter.plugin.common.MethodCall;
 import io.flutter.plugin.common.MethodChannel;
 
 class MethodCallHandlerImpl implements MethodChannel.MethodCallHandler {
 
+  Context context;
+
+  public MethodCallHandlerImpl(Context c) {
+    context = c;
+  }
+
   @Override
   public void onMethodCall(MethodCall call, MethodChannel.Result result) {
     if (call.method.equals("mSupportsAlternateIcons")) {
-      result.success(false);
+      result.success(true);
     } else if (call.method.equals("mGetAlternateIconName")) {
-      result.error("Not supported", "Not supported on Android", null);
+      ActivityInfo activityInfo = IconChanger.getCurrentEnabledIcon(context);
+      result.success(Helper.getIconNameFromActivity(activityInfo.name));
     } else if (call.method.equals("mSetAlternateIconName")) {
-      result.error("Not supported", "Not supported on Android", null);
+      String iconName = call.argument("iconName");
+      if(iconName == null) iconName = "MainActivity";
```

Comment:
MainActivity is not the default value. It is better to use getActivities first element of the array